### DISCUSSION TITLE
Problem connecting to SQL Server using Microsoft driverI am

### DISCUSSION THREAD

**Discussion by TexasGaidheal** [nextgenhealthcare/connect#5209] [Q&A] [answered]
I have developed a channel using Mirth 4.0.1 which reads data from a Sql Server 2008 server and sends the final results as HL7 to a tcp/ip destination. It works perfectly with my test database. It uses the Microsoft jdbc driver. The test database was made as a copy of the live database; the copy was made several years ago.

The details of the jdbc configuration are stored in the configuration map so that the appropriate settings will be used for both the test and live deployments. When I deploy my channel to my live server, it fails when it attempts to access the database. The failure error message appears in the server log; here are the relevant details, which I have edited to improve readability:

```
  CHANNEL:   Quan_Results_JS_DB_Reader
  CONNECTOR: Source
  SOURCE CODE:
  649: var dbServer = configurationMap.get('db_server');
  650: var dbUsername = configurationMap.get('db_username');
  651: var dbPassword = configurationMap.get('db_password');
  652: 
  653: try {
  654:   dbConn = DatabaseConnectionFactory.createDatabaseConnection(dbDriver, dbServer, dbUsername, dbPassword);
  655:   dbQuery = "SELECT TOP 1 " +
  656:               "Cust19_Quan_MCases.CaseID AS ID, " +
  657:               "Cust19_Quan_MCases.AmendmentCount, " +
  658:               "Cust19_Quan_MCases.PrelimStatus, " +
  
  LINE NUMBER:	654
  
  DETAILS:
	Wrapped com.microsoft.sqlserver.jdbc.SQLServerException: The driver could not establish a secure connection to SQL Server by using Secure Sockets Layer (SSL) encryption. 
  Error: "The server selected protocol version TLS10 is not accepted by client preferences [TLS13, TLS12]". ClientConnectionId:b102a2df-3cf0-4fbf-ab0c-7501cc8cfeb0

```
Here are the relevant configuration map settings from my test server. (I have hidden the password):
```
db_driver     com.microsoft.sqlserver.jdbc.SQLServerDriver
db_password   **********
db_server     jdbc:sqlserver://appa-sql-s01:1433;databaseName=CortexTest2
db_username   viaduct
```

The live server has only one different configuration map setting:
```
db_server     jdbc:sqlserver://appa-sql-s01:1433;databaseName=Cortex
```
How can I fix this problem?

**Comment by pacmano1** [comment, marked as answer]
Not really a mirth quesiton, just a jdbc one.   See https://docs.microsoft.com/en-us/sql/connect/jdbc/setting-the-connection-properties?view=sql-server-ver15#properties, search for sslProtocol

**Reply by TexasGaidheal** [reply]
Problem fixed; the Microsoft document reported that different versions of Java could cause changes in behavior of the driver; my live server had a newer version than the test server; I moved the live server back to the older Java version (13) used on the test server and the problem vanished.

**Reply by pacmano1** [reply]
Do make sure you have extra options as per the mirth readme for mcservice/mcserver .vmoptions for java 9+/
